### DISCUSSION TITLE
RNN Doc similarity, example available?

### DISCUSSION THREAD

**Discussion by JulianKingman** [explosion/spaCy#2498] [New Features & Project Ideas]
I saw a recent comment in a blog post that you’re working actively on creating a better doc similarity function, I’d that the RNN method that you mention in this article? 
https://explosion.ai/blog/deep-learning-formula-nlp

Do you have a public example that shows an implementation of that, using user hooks and all that?

Thanks, and thanks for a great Library. I’m just getting started with python, and this lib is so approachable and usable

**Comment by honnibal** [comment]
We don't have an up-to-date example of that. There's the start of an implementation in `spacy.pipeline` --- a class called `SimilarityHook`. If you have a look at the code, you'll see that the `update()` method is unfinished; I think I couldn't figure out what the gold data should look like, and it bothered me that the API was inconsistent with other pipeline components.

The pipeline component should only install the `similarity()` functions override, like the example here does: https://github.com/explosion/spaCy/tree/master/examples/keras_parikh_entailment

If you have time to put these pieces together, a pull request would definitely be welcome. One reason I didn't follow up on this more was that I ended up a bit disappointed by the accuracy and generality of the models. The neural networks take ages to train, and then in the end the results didn't seem to generalise particularly well past the corpus the models were trained on.

**Comment by deepanwayx** [comment]
Hey @honnibal @free-variation Is there a way to obtain the ['entailment', 'contradiction', 'neutral'] label in the parikh_entailment experiment without training the model? With my current resources it is going to take a lot of time to get some reasonable results in the SNLI corpus. Hence I was wondering if I could only use the [demo](https://github.com/explosion/spaCy/blob/master/examples/keras_parikh_entailment/__main__.py#L90) function, without training the model? Are the pretrained model weights and config json file available somewhere which gives >86% in SNLI?
